### PR TITLE
F13: added modal & refactor grid

### PR DESCRIPTION
**Pull request by sushmita12345**
Added Modal component.
Refactor Grid component

Screenshot 2022-03-27 at 4 39 09 PM

### CODE REVIEW

File: assets/css/grid.css [hunk 1]
```diff
@@ -1,15 +1,22 @@
...
-  grid-template-columns: 1fr 1fr;
-  grid-template-rows: 25rem;
+  grid-template-areas:
+    " 2-a 2-b"
+    " 2-c 2-d";
+  padding-left: 10rem;
+  padding-right: 10rem;
 }
 
 .grid-container-1b {
```
Comment (muskanbagrecha) [inline assets/css/grid.css:16]: Maybe the naming can be improved.

File: assets/css/modal.css [hunk 2]
```diff
@@ -0,0 +1,43 @@
+.modal {
+  position: relative;
+  width: 40%;
+  display: none;
+  flex-direction: column;
+  justify-content: center;
+  align-items: center;
+  border: 2px solid black;
```
Comment (muskanbagrecha) [inline assets/css/modal.css]: Avoid using pure colors. Define it in variable.

File: assets/css/modal.css [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  right: 1rem;
+  cursor: pointer;
+}
+
+.modal-content {
+  margin-bottom: 2rem;
+}
+
+.open-modal-btn {
+  background-color: #28282b;
```
Comment (muskanbagrecha) [inline assets/css/modal.css]: Use variables.

File: assets/css/modal.css [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  border: none;
+  color: floralwhite;
```
Comment (muskanbagrecha) [inline assets/css/modal.css]: Avoid pure color.

File: assets/js/modal.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+  openModal.style.display = 'flex';
+  
+});
+
+closeBtn.addEventListener('click', () => {
+  openModal.style.display = 'none';
+});
+
+// window.onclick = function(e) {
```
Comment (muskanbagrecha) [inline assets/js/modal.js]: Remove uncommented code.

File: components/docs.html [hunk 6]
```diff
@@ -1037,24 +1038,67 @@ <h1 class="comp-lib-avatar-color">Simplified Grid</h1>
...
+                            <!-- <div class="grid-container-1a">
                                 <div class="grid-section-1">
                                     <img class="card-img" src="/assets/img/card_image.jpeg" />
                                 </div>
                                 <div class="grid-section-1">
                                     <img class="card-img" src="/assets/img/card_image.jpeg" />
                                 </div>
+                            </div> -->
+
+                            <div class="grid-container-1a">
```
Comment (muskanbagrecha) [inline components/docs.html:1043]: See the naming looks very confusing. Consider renaming.

File: assets/js/modal.js [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+//     if (e.target === openModal) {
+//         openModal.style.display = 'none';
+//     }
+// }
```
Comment (vaishnavipaliya11) [inline assets/js/modal.js]: Remove commented code from you file.

File: components/docs.html [hunk 8]
```diff
@@ -1037,24 +1038,67 @@ <h1 class="comp-lib-avatar-color">Simplified Grid</h1>
...
                             <!-- Two item in grid -->
-                            <div class="grid-container-1a">
+                            <!-- <div class="grid-container-1a">
                                 <div class="grid-section-1">
                                     <img class="card-img" src="/assets/img/card_image.jpeg" />
                                 </div>
                                 <div class="grid-section-1">
                                     <img class="card-img" src="/assets/img/card_image.jpeg" />
                                 </div>
+                            </div> -->
```
Comment (vaishnavipaliya11) [inline components/docs.html]: Remove commented code

**Review by vaishnavipaliya11 — APPROVED** [review]

File: components/docs.html [hunk 9]
```diff
@@ -1295,6 +1339,47 @@ <h3 class="button-sub-heading">Example of Stacked List (Notification Page)</h3>
...
+                            <div class="comp-lib-avatar-content_a">A modal is a dialog box/popup window that is displayed to popup notification on the current page.</div>
+                        
+                            <h3 class="button-sub-heading">Example of Modal</h3>
+
+                            <div>  
+                                <div class="modal-container">
+                                    <button class="open-modal-btn">Open Modal</button>
+                                </div>
+                                <div class="modal-wrapper">
+                                    <div id="modal" class="modal">
```
Comment (muskanbagrecha) [inline components/docs.html:1326]: Why do you need an id and class both?

**Review by muskanbagrecha — APPROVED** [review]
Please implement suggested fixes. Good to merge after that.

**Comment by sushmita12345** [comment]
Changes has been done, thank you.

Screenshot 2022-03-27 at 9 39 01 PM


Screenshot 2022-03-27 at 9 42 38 PM

**Pull request merged**